### PR TITLE
 feat[#4]: Implement JWT authentication. 

### LINKED ISSUE #4: Authentication

### PR DESCRIPTION
**Pull request by JustBeYou**
Fixes #4 

Please merge #23 first.

### CODE REVIEW

File: src/main/java/ro/unibuc/hello/config/JwtTokenFilter.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import java.util.List;
+
+import static io.micrometer.core.instrument.util.StringUtils.isEmpty;
+import static java.util.Optional.ofNullable;
+
+@Component
+@RequiredArgsConstructor
+public class JwtTokenFilter extends OncePerRequestFilter {
+
+    @Autowired
```
[the remark this answers is not in the document]
Comment (JustBeYou) [reply src/main/java/ro/unibuc/hello/config/JwtTokenFilter.java:28]: Copy-pasting has its flaws, I must be honest

File: src/main/java/ro/unibuc/hello/config/JwtTokenUtil.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import java.util.Date;
+
+import static java.lang.String.format;
+
+@Component
+@RequiredArgsConstructor
+public class JwtTokenUtil {
+
+    // TODO: I know this is highly unsecure, but it's just a demo
+    private final String jwtSecret = "very_good_very_nice_secret";
```
[the remark this answers is not in the document]
Comment (JustBeYou) [reply src/main/java/ro/unibuc/hello/config/JwtTokenUtil.java:19]: ❤️

File: src/main/java/ro/unibuc/hello/controller/UserController.java [hunk 3]
```diff
@@ -3,43 +3,22 @@
...
-        } catch (Exception e) {
-            return new ResponseEntity<>(new AuthenticationData(
-                    "",
-                    e.getMessage()
-            ), HttpStatus.BAD_REQUEST);
-        }
-    }
-
-    @GetMapping("/user/{name}")
+    @GetMapping("/{name}")
```
[the remark this answers is not in the document]
Comment (JustBeYou) [reply src/main/java/ro/unibuc/hello/controller/UserController.java:17]: It's impractical to get users by their id. I would keep it like this. One exception isn't so much.

**Review by andreeamatei5810 — APPROVED** [review]
:star2:

**Pull request merged**